llvm/test/CodeGen/NVPTX/jump-table.ll

```diff
@@ -0,0 +1,69 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc < %s | FileCheck %s
+; RUN: %if ptxas %{ llc < %s | %ptxas-verify %}
+
+target triple = "nvptx64-nvidia-cuda"
+
+@out = addrspace(1) global i32 0, align 4
+
+define void @foo(i32 %i) {
```

Comment:
It may be useful to add a test case which triggered the failure in the original patch w/o the barrier attribute.

Reply:
Added